File: JSALGO_AND_DS/Stack_Implementation/Stack.js
class Stack{
    constructor(){
        this.items=[];
    }
    push(item){
        this.items.push(item);
    }
    pop(){
        if(this.items.length==0){
            return "stack is empty || underflow ";
        }
        return this.items.pop();
    }
    display(){
        let str="";
        for(let i=0;i<this.items.length;i++){
            str+= this.items[i]+" ";
        }
        return str;
    }
    isEmpty(){
        return this.items.length==0;
    }
    peek(){
        return this.items[this.items.length-1];
    }
}

let st = new Stack();
st.push(10);
st.push(20);
st.push(30);
st.push(40);
st.push(50);
console.log(st.display());
console.log(st.pop());
console.log(st.pop());
console.log(st.isEmpty());
console.log(st.display());
console.log(st.peek());

File: DSalgo/StackAndQueue/src/stack/Stack.java
package stack;

public class Stack {
	protected int[] data=new int[10];
	protected int tos;
	
	public int pop() throws Exception{
		if(isEmpty()) {
			throw new Exception("Stack is empty");
		}
		int popval = this.data[tos];
		this.data[tos]=0;
		tos--;
		
		return popval;
	}
	public void push(int item) throws Exception{
		if(size()==this.data.length) {
			throw new Exception("Stack is full");
		}
		tos++;
		data[tos]=item;
	}
	public boolean isEmpty(){
		return size()==0;
	}
	public int size() {
		return tos+1;
	}
	public int top() throws Exception{
		if(isEmpty()) {
			throw new Exception("Stack is empty");
		}
		
		return this.data[this.tos];
	}
	
	public void display() {
		for(int i=data.length-1;i>0;i--) {
			System.out.print(data[i]+" ");
		}
	}

}
